### DISCUSSION TITLE
How to use setValue in select filed

### DISCUSSION THREAD

**Discussion by mernaff18** [react-hook-form/react-hook-form#8757] [Q&A]
I tried to set default value form select using setvalue its not working.. how to set value for select filed

   ```
     const fields = ['userEmail', 'roleName'];
         fields.forEach(field => setValue(field, selectedUser[field]));
```

```
<select name="roleName" {...register("roleName", { required: true })} className="form-select" aria-label="Default select example">
                           {allroles?.map((val,index)=>(
                                <option key={index} value={val.roleName}>{val.roleName}</option>
                           ))}  
                        </select>
```

my rolename sample is 

[
    {
        "roleId": 1,
        "roleName": "Product Admin"
    },
    {
        "roleId": 2,
        "roleName": "Member"
    }
]

value is not setting in select .. it always setting first value form the object

**Comment by Moshyfawn** [comment]
Simply the `defaultValue` attribute of your select input?
```jsx
<select defaultValue={allroles[0].roleName} />
```

**Reply by mernaff18** [reply]
setting defaultValue in working.. i need to setValue dynamic in the select field onpage load
